Makefile

```diff
@@ -12,6 +14,8 @@ CMAKE_BUILD_TYPE ?= Debug
 CMAKE_FLAGS := -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE)
 # Extra CMake flags which extend the default set
 CMAKE_EXTRA_FLAGS ?=
+BUILD_DIR ?= build
```

Comment:
not in favor of parameterizing this unless there is a strong use case. Making everything configurable "just because" needs a clean gain, to outweight the costs which are:

- makes build logs and user reports less consistent
- adds churn
- makes the script logic slightly more verbose